clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -64,6 +62,7 @@
   private final String sshKnownHostsFilePath;
   private final boolean sshTrustUnknownHosts;
   private final AuthType authType;
+  private String remoteReference;
```

Comment:
We shouldn't need a class-level property for the remote reference since this will be variable for each time the `download` method is invoked. It looks like you've added it so you have access to it in the archive method. In this case, you might want to resolve the reference inside the `download` method and pass it into the archive method.

Reply:
Noted, let me move that out of the class-level.